workflow/scripts/ts_processors.py

```diff
@@ -169,3 +169,55 @@ def __call__(self, ts: tskit.TreeSequence) -> torch.Tensor:
         
         return sfs
 
+class SPIDNA_processor(BaseProcessor):
+    default_config = {
+        "maf": 0.05,
+        "relative_position": True
+    }
+
+    def __init__(self, config: dict):
+        super().__init__(config, self.default_config)
+
+    def __call__(self, ts: tskit.TreeSequence) -> np.ndarray:
+        # Extract genotype matrix and positions
+        snp = ts.genotype_matrix()  # Shape: (n_variants, n_samples)
+        pos = ts.sites_position
+        
+        # Handle relative positions
+        if self.relative_position:
+            abs_pos = np.array(pos)
+            pos = abs_pos - np.concatenate(([0], abs_pos[:-1]))
+        
+        # MAF filtering
+        if self.maf != 0:
+            num_sample = snp.shape[1]  # Now using shape[1] since matrix isn't transposed
+            if (snp==2).any():
```

Comment:
This is detecting ploidy, right? It's a little messy to do this on the fly (e.g. would fail if there were no homozygous derived alleles, which probably won't happen in practice, but still ...)

A better way is to do this directly from the individuals in the ts, e.g.
```
ploidy = [ind.nodes.size for ind in ts.individuals()]
```

or if we only need the number of haploids
```
samples = ts.num_samples
```

Reply:
this MAF filtering step was part of Yuxin's pipeline that I just copied over. I agree it's a bit messy to do it at this step. let me push a change to use samples from `ts.num_samples`.